pkg/client/unversioned/remotecommand/remotecommand.go

```diff
@@ -194,12 +194,19 @@ func (e *Streamer) doStream() error {
 		if err != nil {
 			return err
 		}
-		defer remoteStdin.Reset()
```

Comment:
It's a bit of a chicken and egg game. With this scenario:

    echo foo | kubectl exec -i <pod> cat

The `cat` command won't terminate until it gets EOF from its stdin. Its stdin comes from the spdy stream. The spdy stream gets its stdin from the non interactive input (`echo foo`). While `foo` is fully read and closed, `doStream` won't exit until `cat` is done, but we aren't resetting `remoteStdin` until `doStream` exits, etc etc etc. This makes sure that `remoteStdin` gets closed/reset as soon as the copy unblocks. Make sense?